### DISCUSSION TITLE
Ignore "old" events on Operator startup

### DISCUSSION THREAD

**Discussion by agardnerIT** [nolar/kopf#1051] [Q&A]
Is it possible to NOT receive events for events older than the startup time. I don't want to "replay" any old events.

The best solution I've come up with so far is:

```
if event['type'] is None:
  print("Old event. Do not process")
  return
```

**Comment by nolar** [comment]
Kopf passes the events from Kubernetes to handlers as is. People might rely on that. 

If you do not need some events, you can add the `when=` callback filter with the condition from your code snippet, and these events will be invisible to handlers (even no logs will be issued).

**Comment by agardnerIT** [comment]
To wrap this up, based on the above feedback, I implemented it [like this](https://github.com/agardnerIT/tracepusher/blob/0ecb9cedc516023a5fdd0052ccddd23741a7e650/operator/operator-logic.py#L172C1-L176C67):

```
@kopf.on.event(
    'pods',
    annotations={"tracepusher/ignore": kopf.ABSENT},
    when=kopf.all_([is_new_event, is_pod_spawned_by_job, has_no_finalizers]))
def on_update_pod(event, spec, name, namespace, logger, **kwargs):
```
